### ISSUE TITLE
🐞 Bug ruby and other programs cause a stack smashing error on ChromeOS 123, limiting chromebrew operation

### ISSUE THREAD

**Issue by noxxwk** [chromebrew/chromebrew#9612]
**Describe the bug**
Ruby and other programs (e.g. Firefox or crew which runs on ruby) cause a stack smashing error and terminates. Very little chromebrew functionality still works.

I tried reinstalling from scratch but the issue remains.

**To Reproduce**

1. I executed this command:

      ```shell
      crew
      ```

      Output:

      ```
      *** stack smashing detected ***: terminated
      Segmentation fault (core dumped)
      ```
     Sometimes you'll get an aborted error:
     ```
     Aborted (core dumped)
      ```

**Expected behavior**
crew and other programs execute.

**Environment: Paste the output of `crew sysinfo -v` between the two `***` below**

crew no longer runs so it's not possible to get the output of `crew sysinfo -v`, but this is a brya device (Asus Vibe CX34) on ChromeOS 123 stable.

I tested and the same issue also occurs on an eve device (Pixelbook).

The behavior began after updating to ChromeOS 123 stable.

**Comment by luhgarlicbread** [comment]
Yeah I've gotten this error too. I just decided to downgrade to ChromeOS v116, so I could have sudo in Crosh still.

**Comment by luhgarlicbread** [comment]
You can downgrade by going on https://cros.tech , searching for your Chromebook's board, and downloading a v116 recovery image.
Flash it to your USB by hitting the gear icon and selecting "use local image" on the Chromebook Recovery Utility extension for Chromium based browsers.

**Comment by yestgoqp9yd1** [comment]
cuz of glibc_235

**Comment by supechicken** [comment]
@yestgoqp9yd1 Oh sorry, I think I accidentally deleted your comment, here is the original comment:
```text
cuz of glibc_235
```

**Comment by supechicken** [comment]
Quick fix:
```shell
unset LD_LIBRARY_PATH
cat /usr/local/etc/crew/meta/glibc_lib235.filelist | grep .so | grep -v gconv | xargs rm -rf
export LD_LIBRARY_PATH=/usr/local/lib64
```

**Comment by VenomousSteam81** [comment]
@supechicken It breaks when you try to install atm, and the second command gives you an error saying that `/usr/local/etc/crew/meta/glibc_lib235.filelist` doesn't exist. To undo that, you would need to run:
```
/usr/bin/sudo /bin/bash 
sudo rm -rf /usr/local/*
```

**Comment by supechicken** [comment]
> atm

You need to have chromebrew installed first. It cannot fix the error during installation.

**Comment by satmandu** [comment]
Expanding on @supechicken 's fix, should we add something line this to /usr/local/etc/profile?

> Quick fix:
> 
```shell
crew_installed_glibc_version=$(/usr/bin/jq --arg key glibc -r '.installed_packages[] | select(.name == $key )| .version ' /usr/local/etc/crew/device.json)
system_installed_glibc_version=$(/usr/local/lib/libc.so.6 | awk -F version 'NR==1{{gsub (" ", "", $0) ; print substr($2, 1, length($2)-1)}}')
if [ "$crew_installed_glibc_version" != "$system_installed_glibc_version" ]; then
  unset LD_LIBRARY_PATH
  grep .so /usr/local/etc/crew/meta/glibc_lib"${crew_installed_glibc_version//.}".filelist | grep -v gconv | xargs rm -rf
  export LD_LIBRARY_PATH=/usr/local/lib64
  crew update ; yes | crew upgrade
fi
```

**Comment by satmandu** [comment]
Thoughts @supechicken @uberhacker @Zopolis4 ?

**Comment by uberhacker** [comment]
This would target only x86_64 systems?  If not, then we should abstract the library path.

**Comment by satmandu** [comment]
The above for inclusion in /usr/local/etc/profile after the paths are set...

and if run standalone:
```
export crew_installed_glibc_version=$(/usr/bin/jq --arg key glibc -r '.installed_packages[] | select(.name == $key )| .version ' /usr/local/etc/crew/device.json)
export installed_glibc_version=$(/lib"$LIB_SUFFIX"/libc.so.6 | /usr/bin/awk -F version 'NR==1{{gsub (" ", "", $0) ; print substr($2, 1, length($2)-1)}}')
if [ "$crew_installed_glibc_version" != "$installed_glibc_version" ]; then
  unset LD_LIBRARY_PATH
  grep .so /usr/local/etc/crew/meta/glibc_lib"${crew_installed_glibc_version//.}".filelist | grep -v gconv | xargs rm -rf
  export LD_LIBRARY_PATH=$CREW_LIB_PREFIX:/lib$LIB_SUFFIX:/usr/lib$LIB_SUFFIX
  crew update ; yes | crew upgrade
fi
```

**Comment by satmandu** [comment]
This should be addressed by the addition of glibc 2.37 to Chromebrew.
